### PR TITLE
Wasm utils vsmt

### PR DESCRIPTION
**Pull request by filiplazovic**
Using Fixed deposit tree from gadgets, removed MerkleClient

Changes to note: Optional block_number, for creating partial notes

Removed functions related to local storage

- [x] Add format version to the note

### CODE REVIEW

File: gadgets/src/smt/smt.rs [hunk 1]
```diff
@@ -25,7 +25,7 @@ pub struct VanillaSparseMerkleTree {
 	pub root: Scalar,
 	curr_index: Scalar,
 	edge_nodes: Vec<Scalar>,
-	leaf_indecies: BTreeMap<ScalarBytes, Scalar>,
+	pub leaf_indecies: BTreeMap<ScalarBytes, Scalar>,
```
Comment (shekohex) [inline gadgets/src/smt/smt.rs:28]:
```suggestion
	pub(crate) leaf_indecies: BTreeMap<ScalarBytes, Scalar>,
```

File: gadgets/src/smt/smt.rs [hunk 2]
```diff
@@ -160,8 +160,8 @@ impl VanillaSparseMerkleTree {
 
 	pub fn prove_zk(
 		&self,
-		k: Scalar,
 		root: Scalar,
+		leaf: Scalar,
 		bp_gens: &BulletproofGens,
 		mut prover: Prover,
```
Comment (shekohex) [inline gadgets/src/smt/smt.rs:166]: is this change in the order of the parameters? with the same type?

File: gadgets/src/fixed_deposit_tree/builder.rs [hunk 3]
```diff
@@ -21,32 +21,49 @@ use sp_std::collections::btree_map::BTreeMap;
...
 		(*r, *nullifier, *nullifier_hash)
 	}
 
 	pub fn prove_zk(
 		&self,
-		k: Scalar,
 		root: Scalar,
+		leaf: Scalar,
 		bp_gens: &BulletproofGens,
 		mut prover: Prover,
```
Comment (shekohex) [inline gadgets/src/fixed_deposit_tree/builder.rs:68]: also here, is this change in the order of the parameters?

**Review by shekohex — COMMENTED** [review]
Sounds good. I left small comments too.

File: gadgets/src/fixed_deposit_tree/builder.rs [hunk 4]
```diff
@@ -61,7 +78,9 @@ impl FixedDepositTree {
 		let mut rng: OsRng = OsRng::default();
 		let mut merkle_proof_vec = Vec::<Scalar>::new();
 		let mut merkle_proof = Some(merkle_proof_vec);
-		let leaf = self.tree.get(k, root, &mut merkle_proof);
+
+		let k = self.tree.leaf_indecies.get(&leaf.to_bytes()).unwrap();
```
Comment (drewstone) [inline gadgets/src/fixed_deposit_tree/builder.rs:82]: `leaf_indices`

File: gadgets/src/smt/smt.rs [hunk 5]
```diff
@@ -59,10 +59,10 @@ impl VanillaSparseMerkleTree {
 	// This function allows this tree to work as a normal tree
 	// Should be deleted in the future if we opt out to use sparse tree
 	// that support non-membership proofs
-	pub fn add(&mut self, vals: Vec<Scalar>) {
+	pub fn add_leaves(&mut self, vals: Vec<[u8; 32]>) {
 		for val in vals {
-			self.update(self.curr_index, val);
-			self.leaf_indecies.insert(val.to_bytes(), self.curr_index);
+			self.update(self.curr_index, Scalar::from_bytes_mod_order(val));
+			self.leaf_indecies.insert(val, self.curr_index);
```
Comment (drewstone) [inline gadgets/src/smt/smt.rs:65]: spelling

File: gadgets/src/smt/smt.rs [hunk 6]
```diff
@@ -171,13 +171,15 @@ impl VanillaSparseMerkleTree {
 		let mut test_rng: OsRng = OsRng::default();
 		let mut merkle_proof_vec = Vec::<Scalar>::new();
 		let mut merkle_proof = Some(merkle_proof_vec);
-		let leaf = self.get(k, root, &mut merkle_proof);
+
+		let k = self.leaf_indecies.get(&leaf.to_bytes()).unwrap();
```
Comment (drewstone) [inline gadgets/src/smt/smt.rs:175]: spelling

**Review by drewstone — CHANGES_REQUESTED** [review]
Looks good, just some spelling issues to fix up before merge.

File: gadgets/src/fixed_deposit_tree/builder.rs (hunk 3 above)
Comment (drewstone) [reply gadgets/src/fixed_deposit_tree/builder.rs:68]: no I think its change in arguments entirely. Before we specified the index of the leaf, now we use the leaf to get the index.

Comment (filiplazovic) [reply gadgets/src/fixed_deposit_tree/builder.rs:68]: Yes, it does require more memory to store all of these indices. I'm yet to figure out a future proof solution to this

Comment (filiplazovic) [reply gadgets/src/fixed_deposit_tree/builder.rs:68]: We can just save indices of the saved leaves (gotten from the notes) as we insert the leaves, but I need to think about how future clients are going to work

**Review by shekohex — APPROVED** [review]
LGTM!

**Pull request merged**